core/src/main/java/io/parsingdata/metal/expression/comparison/ComparisonExpression.java

```diff
@@ -57,7 +59,7 @@ public ComparisonExpression(final ValueExpression value, final ValueExpression p
 
     @Override
     public boolean eval(final ParseGraph graph, final Encoding encoding) {
-        final ImmutableList<Optional<Value>> values = value == null ? ImmutableList.create(Optional.of(graph.current())) : value.eval(graph, encoding);
+        final ImmutableList<Optional<Value>> values = value == null ? ImmutableList.create(graph.current().map(identity())) : value.eval(graph, encoding);
```

Comment:
I think I get it now... This identity is required to convert the `Optional<ParseValue>` to `Optional<Value>`, and the identity method works, even thought it is T -> T, because the variable is defined as `ImmutableList<Optional<Value>>`.

Seriously, it is mind blowing!